### PR TITLE
Remove self-loops and duplicate edges before ParMETIS and restore when converting to DGLGraph

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

fix for https://github.com/dmlc/dgl/issues/3421, https://github.com/dmlc/dgl/issues/3420, https://github.com/dmlc/dgl/issues/3443.
combined from https://github.com/dmlc/dgl/pull/3366, https://github.com/dmlc/dgl/pull/3456.

Verified manually as follows:
1. python3 write_mag.py
2. pm_dglpart mag 2           or          $(which mpirun) --hostfile ~/workspace/ip_config.txt -np 2 pm_dglpart mag 1
3. python3  convert_partition.py --input-dir . --graph-name mag --workspace /tmp/ --output outputs --schema mag.json --num-node-weights 4 --num-parts 2 --removed-edges mag_removed_edges.txt
4. python3 get_mag_data.py
5. python3 verify_mag_partitions.py
6. DGL_HOME=/home/ubuntu/workspace/dgl_toSharedMem DGL_LIBRARY_PATH=$DGL_HOME/build PYTHONPATH=..:.:$DGL_HOME/python:$PYTHONPATH DGLBACKEND=pytorch python3 ~/workspace/dgl_toSharedMem/tools/launch.py --workspace ~/workspace/dgl_toSharedMem/examples/pytorch/rgcn/experimental/ --num_trainers 1 --num_samplers 0 --num_servers 1 --part_config outputs/mag.json --ip_config ~/workspace/ip_config.txt "python3 entity_classify_dist.py --graph-name mag --dataset ogbn-mag --fanout='25,25' --batch-size 1024  --n-hidden 64 --lr 0.01 --eval-batch-size 1024  --low-mem --dropout 0.5 --use-self-loop --n-bases 2 --n-epochs 3 --layer-norm --ip-config ~/workspace/ip_config.txt  --sparse-embedding --sparse-lr 0.06 "

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/rgcn/experimental/README.md [hunk 1]
```diff
@@ -156,6 +156,10 @@ More details about the four steps are explained in our
 The graph structure should be written as a node file and an edge file. The node features and edge features
 can be written as DGL tensors. `write_mag.py` shows an example of writing the OGB MAG graph into files.
 
+As `pm_dglpart` cannot handle self-loops and duplicate edges correctly, these edges are removed and stored
+into `mag_removed_edges.txt`. When converting ParMETIS outputs into DGLGraph in next steps, the file that
```
Comment (zheng-da) [inline examples/pytorch/rgcn/experimental/README.md:160]: i suppose write_mag.py writes the removed edges to mag_removed_edges.txt?

Comment (zheng-da) [reply examples/pytorch/rgcn/experimental/README.md:160]: if so, please mention it in the readme.

Comment (Rhett-Ying) [reply examples/pytorch/rgcn/experimental/README.md:160]: yes, this added sentences lies in `write_mag.py` part.

**Comment by Rhett-Ying** [comment]
Test accuracy is compared between rgcn_dist and rgcn_single with same arguments, namely //example/pytorch/rgcn/experimental/entity_clasify_dist.py VS //examples/pytorch/rgcn/entity_classify_mp.py.

rgcn_dist@2nd epoch: Val Acc 0.4398, Test Acc 0.4252, time: 129.9666
rgcn_single@2nd epoch: Validation Accuracy: 0.4232 | Validation loss: 2.1873 Test Accuracy: 0.4138 | Test loss: 2.2130

cmd of rgcn_dist:
python3 ~/workspace/dgl_toSharedMem/tools/launch.py --workspace ~/workspace/dgl_toSharedMem/examples/pytorch/rgcn/experimental/ --num_trainers 1 --num_samplers 0 --num_servers 1 --part_config outputs/mag.json --ip_config ~/workspace/ip_config.txt "python3 entity_classify_dist.py --graph-name mag --dataset ogbn-mag --fanout='25,25' --batch-size 1024  --n-hidden 64 --lr 0.01 --eval-batch-size 1024 --low-mem --dropout 0.8 --n-bases 2 --use-self-loop --n-epochs 3
--layer-norm --ip-config ~/workspace/ip_config.txt  --sparse-embedding --sparse-lr 0.06 --num_gpus 1 --dgl-sparse"

cmd of rgcn_single:
python3 entity_classify_mp.py --dataset ogbn-mag --fanout='25,25' --batch-size 1024  --n-hidden 64 --lr 0.01 --eval-batch-size 1024  --low-mem --dropout 0.8 --use-self-loop --n-bases 2 --n-epochs 3 --layer-norm --dgl-sparse --sparse-lr 0.06 --gpu 0

**Review by zheng-da — APPROVED** [review]
